### PR TITLE
Fix the memory issues of the VINN branch

### PR DESCRIPTION
**Pull request by dkuegler**
Several optimizations for the VINN eval script, specifically reduces memory consumption of the network!

### CODE REVIEW

**Comment by dkuegler** [comment]
@AhmedFaisal95 or @LeHenschel Someone please double-check on their machine

Ideally, we also run this and the old version to 1-2 subjects and make sure no differences are there (which I do not really expect)
This is mainly due to the part where I don't do the conversion to 320³ from 256³ any more (i.e. this is optimized out for eval), but I also have not changed if this breaks the training due to some unexpected side-effect.

**Comment by dkuegler** [comment]
I had a couple of broken merge things... should be fixed here

**Comment by dkuegler** [comment]
As additional point features, this pull request has better documentation for CUDA out of memory errors #108 , generally reduces the memory footprint by using half-precision during view aggregation #10 , #12

**Comment by LeHenschel** [comment]
Checked the pull request on 80 subjects from the validation cohort (40 1.0 mm, 20 0.8 mm and 20 0.7 mm). Results are consistent with the original version (minor variations due to noise). Only issue is the get_num_classes function is not correctly called, leading to an incorrect shape of the prediction tensor.

File: FastSurferCNN/run_prediction.py [hunk 1]
```diff
@@ -160,71 +169,57 @@ def set_model(self, plane):
...
+    def get_gt(self) -> Tuple[nib.analyze.SpatialImage, np.ndarray]:
         return self.gt, self.gt_data
 
-    def get_orig(self):
+    def get_orig(self) -> Tuple[nib.analyze.SpatialImage, np.ndarray]:
         return self.orig, self.orig_data
 
     def get_prediction(self):
-        oh, ow, od = self.orig.shape
+        shape = self.orig.shape + (self.model.get_num_classes(),)
```
Comment (LeHenschel) [inline FastSurferCNN/run_prediction.py:185]: this needs to be changed to shape = self.orig.shape + (self.get_num_classes(),)

**Review by LeHenschel — APPROVED** [review]

**Pull request merged**